provisioning/transport/amqp/src/ProvisioningTransportHandlerAmqp.cs

```diff
@@ -106,19 +135,21 @@ public class ProvisioningTransportHandlerAmqp : ProvisioningTransportHandler
                 string linkEndpoint = $"{message.IdScope}/registrations/{registrationId}";
 
                 using AmqpClientConnection connection = authStrategy.CreateConnection(builder.Uri, message.IdScope);
-                await authStrategy.OpenConnectionAsync(connection, s_timeoutConstant, useWebSocket, Proxy, RemoteCertificateValidationCallback).ConfigureAwait(false);
```

Comment:
```s_timeoutConstant``` was the hardcoded 1 minute timespan that we passed into the AMQP library to use in the place of a cancellation token. This PR allows users to control what that timespan is configured to instead of this hardcoded value